core/templates/pages/topic-editor-page/services/topic-editor-state.service.ts

```diff
@@ -18,541 +18,556 @@
-       */
-      updateExistenceOfTopicName: function(topicName, successCallback) {
-        EditableTopicBackendApiService.doesTopicWithNameExistAsync(
-          topicName).then(
-          function(topicNameExists) {
-            _setTopicWithNameExists(topicNameExists);
-            if (successCallback) {
-              successCallback();
-            }
-            $rootScope.$applyAsync();
-          }, function(error) {
-            AlertsService.addWarning(
-              error ||
+  /**
+   * Attempts to set the boolean variable _topicWithNameExists based
+   * on the value returned by doesTopicWithNameExistAsync and
+   * executes the success callback provided. No arguments are passed to the
+   * success callback. Execution of the success callback indicates that the
+   * async backend call was successful and that _topicWithNameExists
+   * has been successfully updated.
+   */
+  updateExistenceOfTopicName(
+      topicName: string,
+      successCallback: Function): void {
+    this.editableTopicBackendApiService.doesTopicWithNameExistAsync(
+      topicName).then(
+      (topicNameExists) => {
+        this._setTopicWithNameExists(topicNameExists);
+        if (successCallback) {
+          successCallback();
+        }
+      }, (error) => {
+        this.alertService.addWarning(
+          error ||
               'There was an error when checking if the topic name ' +
               'exists for another topic.');
-          });
-      },
+      });
+  }
 
-      /**
-       * Attempts to set the boolean variable _topicWithUrlFragmentExists based
-       * on the value returned by doesTopicWithUrlFragmentExistAsync and
-       * executes the success callback provided. No arguments are passed to the
-       * success callback. Execution of the success callback indicates that the
-       * async backend call was successful and that _topicWithUrlFragmentExists
-       * has been successfully updated.
-       */
-      updateExistenceOfTopicUrlFragment: function(
-          topicUrlFragment, successCallback) {
-        EditableTopicBackendApiService.doesTopicWithUrlFragmentExistAsync(
-          topicUrlFragment).then(
-          function(topicUrlFragmentExists) {
-            _setTopicWithUrlFragmentExists(topicUrlFragmentExists);
-            if (successCallback) {
-              successCallback();
-            }
-            $rootScope.$applyAsync();
-          }, function(error) {
-            AlertsService.addWarning(
-              error ||
+  /**
+   * Attempts to set the boolean variable _topicWithUrlFragmentExists based
+   * on the value returned by doesTopicWithUrlFragmentExistAsync and
+   * executes the success callback provided. No arguments are passed to the
+   * success callback. Execution of the success callback indicates that the
+   * async backend call was successful and that _topicWithUrlFragmentExists
+   * has been successfully updated.
+   */
+  updateExistenceOfTopicUrlFragment(
```

Comment:
This does an async operation and therefore we had used a `$rootScope.$applyAsync();`. but that has been removed. How are we handling this now?

Reply:
Converted the function to return a promise and every angularJS file where this function was called, I have added an `$rootScope.$applyAsync();`. This removes the requirement of using `$rootScope.$applyAsync();` in the function itself